src/main.rs

```diff
@@ -1,3 +1,141 @@
+#[macro_use]
+extern crate clap;
+
+mod options;
+use crate::options::CLIOptions;
+use clap::App;
+use std::{
+    fs,
+    fs::File,
+    io,
+    path::{Path, PathBuf},
+};
+
 fn main() {
-    println!("Hello, world!!!");
+    let yaml = load_yaml!("../cli_def/en_us.yml");
+    let matches = App::from_yaml(yaml).get_matches();
+
+    let raw_options = CLIOptions::from_clap_matches(&matches);
+
+    let options = raw_options.process_options();
+
+    for filename in options.files {
+        let file = File::open(filename).unwrap();
+        unzip_file(file, &PathBuf::from(&raw_options.destination_folder));
+    }
+}
+
+fn unzip_file(file: File, directory: &Path) {
```

Comment:
Maybe we can export this function? We can make the `main.rs` only calling all necessary things and the `lib.rs` can have publical accessible functions/structs/traits and etc.

Reply:
True! 
Let me split both.